DatabaseSnapshot.cpp

```diff
@@ -27,6 +57,31 @@ const OnDiskDataset *DatabaseSnapshot::find_dataset(const std::string &name) con
     return nullptr;
 }
 
+
+void DatabaseSnapshot::read_iterator(
+    Task *task,
+    const std::string &iterator_id,
+    int count,
+    std::vector<std::string> *out
+) const {
+    auto it = iterators.find(iterator_id);
+    if (it == iterators.end()) {
+        throw std::runtime_error("tried to read non-existent iterator");
+    }
+
+    // TODO maybe should busy-wait on failure?
+    db_handle.request_iterator_lock(iterator_id);
```

Comment:
Doesn't it mean that right now concurrent iterator access will fail with an error and the user has to be aware of that?

Reply:
as far as I see this is not handled right now in https://github.com/CERT-Polska/mquery/pull/59/files